internal/html/form.go

```diff
@@ -161,6 +163,8 @@ var inputFieldSnippet = NewSnippet(`
 			{{ if .Spec.AutoFocus }}autofocus{{ end }}
 			class="row-input {{if .State.ErrorMessage}}form-error{{end}}"
 			autocomplete="{{if .Spec.AutocompleteMode}}{{.Spec.AutocompleteMode}}{{else}}off{{end}}"
+			{{if .Spec.InputMode}}inputmode="{{.Spec.InputMode}}"{{end}}
+			{{if .Spec.Pattern}}pattern="{{.Spec.Pattern}}"{{end}}
```

Comment:
Those are not strictly required but should make the mobile experience and autofill better.